plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -0,0 +1,389 @@
+import static io.trino.plugin.jdbc.StandardColumnMappings.shortDecimalWriteFunction;
+import static io.trino.plugin.jdbc.StandardColumnMappings.smallintColumnMapping;
+import static io.trino.plugin.jdbc.StandardColumnMappings.smallintWriteFunction;
+import static io.trino.plugin.jdbc.StandardColumnMappings.timestampColumnMapping;
+import static io.trino.plugin.jdbc.StandardColumnMappings.tinyintColumnMapping;
+import static io.trino.plugin.jdbc.StandardColumnMappings.tinyintWriteFunction;
+import static io.trino.plugin.jdbc.StandardColumnMappings.varbinaryWriteFunction;
+import static io.trino.plugin.jdbc.StandardColumnMappings.varcharColumnMapping;
+import static io.trino.plugin.jdbc.StandardColumnMappings.varcharWriteFunction;
+import static io.trino.spi.StandardErrorCode.NOT_SUPPORTED;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.BooleanType.BOOLEAN;
+import static io.trino.spi.type.DateType.DATE;
+import static io.trino.spi.type.DecimalType.createDecimalType;
+import static io.trino.spi.type.DoubleType.DOUBLE;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.RealType.REAL;
+import static io.trino.spi.type.SmallintType.SMALLINT;
+import static io.trino.spi.type.TinyintType.TINYINT;
+import static io.trino.spi.type.VarcharType.createUnboundedVarcharType;
+import static java.lang.Integer.min;
+import static java.lang.String.format;
+import static java.lang.String.join;
+import static java.util.Locale.ENGLISH;
+
+/**
+ * The problem of incorrect catalog names kept tormenting me, I kept modifying it, overriding methods,
+ * and reading the source code in depth, but the problem resided in my head like a ghost.
+ * I once suspected that it was the problem of ClickHouse jdbc driver , and tried to replace another
+ * driver package during the period, but the problem remained.
+ * No matter how I configure it, the name of the catalog will always become the default database of
+ * clickhouse (default), which will cause all the values obtained by `connection.getCatalog()` are not fought for.
+ * <p>
+ * In the end, I had to override all the methods where catalog was wrong
+ */
+public class ClickHouseClient
+        extends BaseJdbcClient
+{
+    private static final Logger log = Logger.get(ClickHouseClient.class);
+
+    @Inject
+    public ClickHouseClient(BaseJdbcConfig config, ConnectionFactory connectionFactory, TypeManager typeManager)
+    {
+        super(config, "\"", connectionFactory);
+    }
+
+    @Override
+    protected String quoted(@Nullable String catalog, @Nullable String schema, String table)
+    {
+        StringBuilder sb = new StringBuilder();
+        if (!isNullOrEmpty(schema)) {
+            sb.append(quoted(schema)).append(".");
+        }
+        sb.append(quoted(table));
+        return sb.toString();
+    }
+
+    @Override
+    public String quoted(RemoteTableName remoteTableName)
+    {
+        return quoted(
+                null,
```

Comment:
if you apply my proposed fix for https://github.com/trinodb/trino/pull/4909/files?file-filters%5B%5D=.java&file-filters%5B%5D=.xml#r562287114, you won't need to override this either